projects/plugins/jetpack/modules/subscriptions/subscribe-modal/class-jetpack-subscribe-modal.php

```diff
@@ -217,10 +217,10 @@ public function should_user_see_modal() {
 			return false;
 		}
 
-		// Don't show if subscribe query param is set.
-		// It is set when user submits the subscribe form.
+		// Don't show if one of subscribe query params is set.
+		// They are set when user submits the subscribe form.
 		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
-		if ( isset( $_GET['subscribe'] ) ) {
+		if ( isset( $_GET['subscribe'] ) || isset( $_GET['blogsub'] ) ) {
```

Comment:
Tested well with Jetpack self hosted, which gave URLs like`?subscribe=pending#subscribe-blog-3`

Looking at the code  `blogsub` is coming from the wp.com simple backend for paid subscriptions only?